client/my-sites/patterns/index.tsx

```diff
@@ -0,0 +1,33 @@
+import DocumentHead from 'calypso/components/data/document-head';
+import Main from 'calypso/components/main';
+import { PatternsCategoryBody } from 'calypso/my-sites/patterns/bodies/category';
+import { PatternsHomeBody } from 'calypso/my-sites/patterns/bodies/home';
+import type { PatternGalleryFC } from 'calypso/my-sites/patterns/types';
+
+type PatternsHomePageProps = {
+	category: string;
+	isGridView?: boolean;
+	patternGallery: PatternGalleryFC;
+};
+
+export const PatternsIndex = ( {
+	category,
+	isGridView,
+	patternGallery: PatternGallery,
+}: PatternsHomePageProps ) => {
+	return (
+		<Main isLoggedOut fullWidthLayout>
+			<DocumentHead title="WordPress Patterns" />
+
+			{ ! category ? (
+				<PatternsHomeBody />
+			) : (
+				<PatternsCategoryBody
+					category={ category }
+					isGridView={ isGridView }
+					patternGallery={ PatternGallery }
+				/>
```

Comment:
I'm not a big fan of the `Body` suffix, although I see where you're coming from.
```suggestion
				<PatternsHomePage />
			) : (
				<PatternsCategoryPage
					category={ category }
					isGridView={ isGridView }
					patternGallery={ PatternGallery }
				/>
```

That would also allow us to put them under a `pages` folder rather than a `bodies` folder (which sounds a bit strange)